pkg/controller/node/ipam/cloud_cidr_allocator.go

```diff
@@ -146,7 +146,10 @@ func (ca *cloudCIDRAllocator) worker(stopChan <-chan struct{}) {
 				glog.Warning("Channel nodeCIDRUpdateChannel was unexpectedly closed")
 				return
 			}
-			ca.updateCIDRAllocation(workItem)
+			if err := ca.updateCIDRAllocation(workItem); err != nil {
+				// Requeue the failed node for update again.
+				ca.insertNodeToProcessing(workItem)
```

Comment:
This is incorrect. Nothing is really picking nodes from nodesInProcessing set.

Reply:
Nice catch.. I confused nodesInProcessing for the nodeUpdate channel. Fixed it now.